instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/idling/RouteAlternativesIdlingResource.kt

```diff
@@ -52,14 +53,19 @@ class RouteAlternativesIdlingResource(
         alternatives: List<DirectionsRoute>,
         routerOrigin: RouterOrigin
     ) {
+        if (waifForNonEmptyAlternativesList && alternatives.isEmpty()) {
+            mapboxNavigation.setNavigationRoutes(mapboxNavigation.getNavigationRoutes().take(1))
+            return
+        }
         // Verify this happens on the main thread.
-        if (Looper.myLooper() != Looper.getMainLooper()) {
-            calledOnMainThread = false
+        if (Looper.myLooper() == Looper.getMainLooper()) {
+            calledOnMainThread = true
```

Comment:
just curious, initial option didn't work for some case, did it?

Reply:
it worked, make it reverted to avoid check passed if `calledOnMainThread` is not called.  Ideally there should be an additional check something like `wasInvoked`, but it's enough for now I think